src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol_ImplementationChecks.cs

```diff
@@ -995,8 +995,23 @@ void checkSingleOverriddenMember(Symbol overridingMember, Symbol overriddenMembe
                                 }
                                 else
                                 {
-                                    // error CS0508: return type must be 'C<V>' to match overridden member 'M<T>()'
-                                    diagnostics.Add(ErrorCode.ERR_CantChangeReturnTypeOnOverride, overridingMemberLocation, overridingMember, overriddenMember, overriddenMethod.ReturnType);
+                                    if (getInvalidNullableTypeArgument(overridingMethod.ReturnType) is { } typeParameter &&
+                                        (overridingMember as SourceOrdinaryMethodSymbol)?.GetSyntax().ConstraintClauses.Count == 0)
+                                    {
+                                        // error CS8971: '{0}': return type must be '{2}' to match overridden member '{1}'. Consider adding a 'where {3} : {4}' constraint to the overridden member type parameter.
```

Comment:
```suggestion
                                        // error CS8971: '{0}': return type must be '{2}' to match overridden member '{1}'. Consider adding a 'where {3} : {4}' constraint.
```

I think there's not a need to mention the fact that the constraint applies to a particular type parameter. The example constraint syntax itself shows this to the reader. Also, I find 'overridden member' misleading here because we're referring to a change that needs to be made on the *overriding* member.